### ISSUE TITLE
Inconsistent state when disabling "error_pages"

### ISSUE THREAD

**Issue by Prinzhorn** [apex/up#735]
## Prerequisites

* [x] I am running the latest version. (`up upgrade`)
* [x] I searched to see if the issue already exists.
* [x] I inspected the verbose debug output with the `-v, --verbose` flag.
* [x] Are you an Up Pro subscriber?

## Description

Despite

```json
"error_pages": {
  "disable": true
},
```

the error page exists and the originally configured `support_email` still lingers around.

## Steps to Reproduce

Originally I configured this

```json
"error_pages": {
  "variables": {
    "support_email": "example@example.com"
  }
},
```

later I switched to

```json
"error_pages": {
  "disable": true
},
```

Now much later I wanted to add a domain to production. I had to ctrl+c the `up stack apply` process because I hadn't configured e-mail yet and it was waiting for confirmation for the certificate. However, after killing it the deployment was broken (likely due to my user error) but I was greeted with

![selection_375](https://user-images.githubusercontent.com/679144/48359842-1c5edb80-e69e-11e8-9c51-63afcca9a410.png)

So even though I disabled the error pages weeks ago, the e-mail address still lingers somewhere and the error page was shown. How is that possible?

This was unfortunately the second time I was able to get up into an inconsistent state, but I didn't open an issue last time.

**Comment by tj** [comment]
The error page config is updated along with the Lambda (since it's in the Up proxy code itself), may need to re-deploy instead of `up stack apply`, but if it the problem persists after that let me know!

**Comment by Prinzhorn** [comment]
Like I said I disabled the error pages weeks ago and they were disabled. Let me rephrase what this issue is about: why is the original `support_email` still stored somewhere? I'm just trying to understand where it came from.

**Comment by tj** [comment]
It's only stored in the up.json, definitely should disappear if it was removed, I'll try and reproduce that

**Comment by tj** [comment]
No luck yet, with 0.7.8-pro I tried deploying with it enabled, then again disabled a few times but it seems ok. Up's proxy just reads from ./up.json on disk, so as long as it's removed I don't know how it could possibly get back in

**Comment by Prinzhorn** [comment]
> No luck yet, with 0.7.8-pro I tried deploying with it enabled, then again disabled a few times but it seems ok. Up's proxy just reads from ./up.json on disk, so as long as it's removed I don't know how it could possibly get back in

Thank, I'll see if I can reproduce it.

> Up's proxy just reads from ./up.json on disk

If that's the case then the only explanation is that the inconsistent deploy I had somehow resulted in a (much) older deployment being invoked. I'm not sure how rollbacks etc work, does the old stuff still sit somewhere?

**Comment by Prinzhorn** [comment]
I was able to reproduce it:

1. Clone https://github.com/Prinzhorn/up-issue-735
2. Adjust "profile"
3. `up`
4. Open "/404" to see the error page with "support@example.com"
5. Disable `error_pages`
6. `up`
7. Open "/404" to see the default express error handler page "Cannot GET /404"
8. Add a domain to staging `"stages": { "staging": { "domain": "example.com" } }`
9. `up stack plan`
10. `up stack apply`
11. `up stack status` (something like "DnsZoneExampleCom: example.com (Service: AmazonRoute53; Status Code: 400; Error Code: InvalidDomainName")
12. Open any staging URL (e.g. "/404") and see the same page as in step 4

**Comment by s32x** [comment]
Any update on this? I've actually been experiencing a similar issue... However I don't think it's related to the `error_pages=true` flag in particular. I believe the lack of change in the error page is just a really obvious case of the app not updating properly since I've noticed my `up` deploys for new code aren't persisting. Cloudwatch logs say I'm running a commit hash from a week ago which is the hash from my initial up deploy. 

Is `up` not the command for deploying new revisions?

**Comment by s32x** [comment]
I did some more playing and noticed that when I hit the `execute-api` url that up spits out it returns the desired output and cloudwatch gives me the most recent hash. Hitting my custom domain again returns the old codes output and cloudwatch has the old hash. I'm super new to aws/lambda so I'm no expert on this stuff but it feels like API gateway has something to do with it...

As a matter of fact from what it looks like API gateway is bound to my `production` environment.. whereas I believe `up` deploys updates to my `staging` environment. It looks like I have just found my problem :) `up production` promoted my code to prod and everything is good on my custom domain. Hopefully, this helps others. I guess I'm just having a tough time differentiating environments in a serverless architecture and just lambda in general.

Thanks so much for this awesome application! Makes my life 1m times easier.
